Content.Server/MachineLinking/Components/SignalTimerComponent.cs

```diff
@@ -0,0 +1,38 @@
+using Content.Shared.MachineLinking;
+using Robust.Shared.Audio;
+using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom.Prototype;
+
+namespace Content.Server.MachineLinking.Components
+{
+    /// <summary>
+    ///     This is a modified version of the signal switch.
+    /// </summary>
+
+    [RegisterComponent]
+    public sealed class SignalTimerComponent : Component
+    {
+        [DataField("length")]
+        public float Length;
```

Comment:
You should probably use timespans instead of floats for this and time remaining

Reply:
Sure. I think I still have to store float length though, so that the TargetTime can be set whenever the timer is activated (and not when the value is set)?